apps/browser-extension-wallet/src/components/MainMenu/DropdownMenuOverlay/components/WalletAccounts.tsx

```diff
@@ -62,20 +75,39 @@ export const WalletAccounts = ({ isPopup, onBack }: { isPopup: boolean; onBack:
   );
 
   const unlockAccount = useCallback(
-    async (accountIndex: number) => {
-      await addAccount({
-        wallet,
-        accountIndex,
-        metadata: { name: defaultAccountName(accountIndex) }
-      });
+    async (accountIndex: number) =>
+      wallet.type === WalletType.InMemory
+        ? enableAccountPasswordPrompt.open({ accountIndex, password: '' })
+        : await addAccount({
```

Comment:
We also need a popup for instructions to connect hardware device
```suggestion
        : enableAccountHardwarePrompt.open({ accountIndex, type: wallet.type })
```